src/colorizer/loaders/workers/urlLoadWorker.ts

```diff
@@ -0,0 +1,103 @@
+  data: number[] | boolean[];
+  min?: number;
+  max?: number;
+};
+
+type LoadedData<T extends FeatureDataType> = {
+  data: FeatureArrayType[T];
+  min: number;
+  max: number;
+};
+
+async function loadFromJsonUrl(url: string, type: FeatureDataType): Promise<LoadedData<typeof type>> {
+  const result = await fetch(url);
+  const text = await result.text();
+  // JSON does not support `NaN` so we use `null` as a placeholder for it, then convert back
+  // to `NaN` when parsing the data.
+  const parseResult: FeatureDataJson = JSON.parse(nanToNull(text));
+  let { data: rawData } = parseResult;
+  const { min, max } = parseResult;
+
+  for (let i = 0; i < rawData.length; i++) {
+    if (rawData[i] === null) {
+      rawData[i] = NaN;
+    }
+  }
+
+  if (isBoolArray(rawData)) {
+    rawData = rawData.map(Number);
+  }
+
+  // If min/max is not provided, calculate it from the data
+  let dataMin = Number.POSITIVE_INFINITY;
+  let dataMax = Number.NEGATIVE_INFINITY;
+  for (let i = 0; i < rawData.length; i++) {
+    dataMin = Math.min(rawData[i], dataMin);
+    dataMax = Math.max(rawData[i], dataMax);
+  }
+
+  // Construct typed array from raw data so it can be transferred w/o copying to
+  // main thread
+  const data = new featureTypeSpecs[type].ArrayConstructor(rawData);
+
+  return { data, min: min ?? dataMin, max: max ?? dataMax };
+}
+
+async function loadFromParquetUrl(url: string, type: FeatureDataType): Promise<LoadedData<typeof type>> {
+  const result = await fetch(url);
+  const arrayBuffer = await result.arrayBuffer();
+  let data: FeatureArrayType[typeof type] = new featureTypeSpecs[type].ArrayConstructor(0);
+  let dataMin: number = Number.POSITIVE_INFINITY;
+  let dataMax: number = Number.NEGATIVE_INFINITY;
+
+  await parquetRead({
+    file: arrayBuffer,
+    compressors,
+    onComplete: (rawData: number[][]) => {
+      const flattenedMap = rawData.flat().map((value) => {
+        return value === null ? NaN : Number(value);
+      });
+      data = new featureTypeSpecs[type].ArrayConstructor(flattenedMap);
+      // Get min and max values for the data
+      for (let i = 0; i < data.length; i++) {
+        const value = Number(data[i]);
+        dataMin = dataMin === undefined ? value : Math.min(dataMin, value);
+        dataMax = dataMax === undefined ? value : Math.max(dataMax, value);
+      }
+    },
+  });
+  return { data, min: dataMin, max: dataMax };
+}
+
+async function load(url: string, type: FeatureDataType): Promise<Transfer> {
+  let result: LoadedData<typeof type>;
+  if (url.endsWith(".json")) {
+    result = await loadFromJsonUrl(url, type);
+  } else if (url.endsWith(".parquet")) {
+    result = await loadFromParquetUrl(url, type);
+  } else {
+    throw new Error(`Unsupported file format: ${url}`);
+  }
+
+  const { min, max, data } = result;
+
+  // TODO: Also generate textures for the data on the worker thread too?
+  // Would need access to the underlying array buffer to transfer
```

Comment:
I'm going to make a follow-up PR that does this! This one is getting a little unwieldy already.

Reply:
how much savings will there be in creating the textures on the worker?  If this is not necessary then why bother?  Save it for later?

Reply:
This was a good reminder for me to go back and validate my claims 😄

![image](https://github.com/user-attachments/assets/7906f92f-3997-4a9c-98bd-23f76d68553d)

It's adding about 2-3 milliseconds each time a feature is loaded. It's small relative to the React re-render but it's there. I think it will be a relatively small lift though.